lib/vdsm/storage/lvm.py

```diff
@@ -577,6 +577,73 @@ def _getVGDevs(self, vgNames):
                 devices = tuple(devices)
         return devices
 
+    def _updatevgs(self, vgs_output, vg_names):
+        """
+        Update cached VGs based on the output of the LVM command.
+        Remove stale VGs (have not been updated) out of the 'vg_names' list,
```

Comment:
This is little bit confusing - we don't remove names from vg_names list.

Describing the side effects of this function can be something like this:

- Add new vgs to the cache
- Replace vgs in the cache with vgs reported by the vgs command, updating the vg attributes.
- If called without vg names, remove all vgs from the cache not reported by lvm
- If called with vg names, remove specifed vgs from the cache if they were not reported by lvm.

I think the best way to describe this is to have tests verifying the contents of the cache after the call with all cases. Using fake runner with fake lvm output is the easier way to test this. However since this patch just simple move, I don't think we need to spend time on the tests now, unless the code you are moving is not covered by the tests.